docs/platform_developers/developer_getting_started.rst

```diff
@@ -0,0 +1,211 @@
+
+.. code-block:: bash
+
+    pip install -r requirements.txt
+
+    pip install -r requirements.dev.txt
+
+
+Set project sources
+#####################
+
+When starting a new project, the next step is to set the sources of your project so that it knows, where to get its required modules from. Otherwise, you can skip this step and just ``import`` your desired modules.
+
+If you only use opensource modules as provided by Inmanta, you can skip below step. 
+
+1. Find the module you want to work on
+2. Copy the SSH URL of the repo
+3. In your VS code, open the ``project.yml`` file and under ``repo:``, add the copied line there but keep in mind to replace the name of a specific module with a place holder, like below example:
+
+.. code-block:: bash
+
+    code project.yml
+
+.. code-block:: yaml
+    
+    repo:
+        - git@code.inmanta.com:example/my_module.git
+
+Becomes:
+
+.. code-block:: yaml
+    
+    repo:
+        - git@code.inmanta.com:example/{}.git
+
+* Now, in your ``main.cf`` file, if you import a module like, ``import <my_module>`` and save the file, you can get code completion. If you are working on an exisitng project with a populated ``main.``cf file, code completion will work as expected.
+
+**Please note, code completion and navigation work on modules that are imported in the ``main.cf`` file**.
+
+
+Module developers guide
+#########################
+
+Like projects, there are also two scenarios:
+
+1. Working on a new module :ref:`Working on a New Module`.
+2. Working on an existing module :ref:`Working on an Existing Module`.
+
+
+Working on a New Module
+=======================
+
+There is a guide `here <https://docs.inmanta.com/community/latest/model_developers/modules.html>`_ that helps you get up and running.
+
+Same as :ref:`Working on a New Project` part, modules can also be created like:
+
+.. code-block:: bash
+
+    pip install cookiecutter
+
+    cookiecutter https://github.com/inmanta/inmanta-module-template.git
+
+
+For more details go `here <https://github.com/inmanta/inmanta-module-template>`_ and `here <https://docs.inmanta.com/community/dev/model_developers/modules.html>`_.
+
+
+Working on an Existing Module
+=============================
+
+When starting to work on an existing module, it is recommended to check the ``readme.md`` file that comes with the module to see the instructions on how to install and use them. There is also a guide `here <https://docs.inmanta.com/community/latest/model_developers/modules.html>`_ that is useful in case you skipped the previous part.
+
+
+Required Environment Variables
+##############################
+
+There are a few environment variables that are required whenever you need to do unit tests to indicate the device, port, username and password, etc. which are mentioned here.
+
+It is also *recommended* to set the ``INMANTA_TEST_ENV`` environment variable to speed up your tests and avoid creating virtual environments at each test run. It can be set to something like:
```

Comment:
```suggestion
It is *recommended* to set the ``INMANTA_TEST_ENV`` environment variable to speed up your tests and avoid creating virtual environments at each test run. It can be set to something like:
```